src/test/java/com/superdevs/HealthOMeter/service/calculatorServiceTests/CalculatorServiceTestsForBMI.java

```diff
@@ -0,0 +1,99 @@
+package com.superdevs.HealthOMeter.service.calculatorServiceTests;
+
+import com.superdevs.HealthOMeter.calculators.BMICalculator;
+import com.superdevs.HealthOMeter.calculators.RFMCalculator;
+import com.superdevs.HealthOMeter.calculators.WHRCalculator;
+import com.superdevs.HealthOMeter.service.CalculatorService;
+import com.superdevs.HealthOMeter.service.managers.CalculatorManager;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.math.BigDecimal;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@SpringBootTest
+@RunWith(SpringRunner.class)
+public class CalculatorServiceTestsForBMI {
+
+    public ApplicationContext context = new AnnotationConfigApplicationContext(CalculatorManager.class,
+            CalculatorService.class, BMICalculator.class);
+
+    public CalculatorService calculatorService = context.getBean(CalculatorService.class);
+
+
+    @Test
+    public void testCalculatorServiceForBMI() {
+
+        //Given
+        BigDecimal weight = new BigDecimal(119.522);
+        BigDecimal height = new BigDecimal(197);
+        BigDecimal expectedValue = new BigDecimal(30.8);
```

Comment:
The results of this constructor can be somewhat unpredictable. It is better to use BigDecimal(String) contructor, which is predictable.

Reply:
Because BigDecimal(double) contruktor was used we get values: 
Expected :30.800000000000000710542735760100185871124267578125
Actual   :30.8

When we use BigDecimal(String) then then the test will pass.

Reply:
The rest of the tests in which BigDecimal(double) constructor was used passes, because we do not compare created values. We check cases where weight/height = 0, so we expect value 0 to be returned.